### DISCUSSION TITLE
Why raising UserWarning is logging from on_validation_epoch_end and only in rank0

### DISCUSSION THREAD

**Discussion by pengzhenghao** [Lightning-AI/pytorch-lightning#17354] [DDP / multi-GPU / multi-node]
I am trying to log from on_validation_epoch_end  and only in rank 0:

```python
    def on_validation_epoch_end(self):
        # get some data...
        if self.trainer.is_global_zero:
            # do some eval
            self.log_dict(result, rank_zero_only=True)
```

If I understand correctly, I am not suppose to set `rank_zero_only=True` with `sync_dist=True` together because they are conflicting.

However, I got the below warning message flushing the screen.

```
PossibleUserWarning: It is recommended to use `self.log('eval/XXX', ..., sync_dist=True)` when logging on epoch level in distributed setting to accumulate the metric across devices
```

So would that be a issue if I don't do `sync_dist=True`? If that's OK, then anyone know how to suppress the UserWarning? Thanks！

**Comment by ZhaoGe666** [comment]
I met exactly the same issue as yours, all we can do by far is to suppress it. Here is an example:
https://github.com/Lightning-AI/pytorch-lightning/issues/10182#issuecomment-953042541
